client/src/panels/panel_declarations/finances/transfer_payments/tp_by_region.js

```diff
@@ -11,81 +11,130 @@ import {
+  const load_csv = () => make_request( get_static_url(`csv/population.csv`) )
+    .then( csv_string => parse_csv_string(csv_string) );
+    
+  return load_csv().then(function(loaded) {
+    const transformed = _.chain(loaded)
+      .keyBy(row => row[0])
+      .mapValues(row => _.chain(row)
+        .tail()
+        .map(value => parseInt(value))
+        .value()
+      )
+      .value();
+    
+    return transformed;
   }
+  ); 
+};
+
 
-  const data = std_years.map( year => table.prov_code(year, level === "gov" ? false : subject.id) );
+const prepare_data_for_a11y_table = (data) => _.chain(data)
+  .flatMap( _.keys )
+  .uniq()
+  .map( (prov_code) => {
+    const formatted_data = _.map(
+      data,
+      (row) => formats["compact2_written_raw"](row[prov_code] || 0)
+    );
   
-  const max = _.chain(data)
-    .last()
-    .values()
-    .max()
-    .value();
-  const color_scale = d3.scaleLinear()
-    .domain([0, max])
-    .range([0.2, 1]);
-
-  return max > 0 && {
-    data,
-    color_scale,
-    years: std_years,
-    formatter,
-  };
-};
+    return {
+      label: provinces[prov_code].text,
+      data: formatted_data,
+    };
+  } )
+  .filter('data')
+  .value();
 
-const prepare_data_for_a11y_table = (data) =>
-  _.chain(provinces)
-    .map((province, prov_code) => {
-      if( !_.includes(["onlessncr", "qclessncr", "ncr"], prov_code)){
-        const formatted_yearly_tp = _.map(
-          data, 
-          (row) => formats["compact2_written_raw"](row[prov_code])
-        );
-
-        return {
-          label: province.text,
-          data: formatted_yearly_tp,
-        };
-      }
-    })
-    .filter((data) => !_.isUndefined(data))
-    .value();
 
-export const declare_tp_by_region_panel = () => declare_panel({
-  panel_key: "tp_by_region",
-  levels: ["gov", "dept"],
-  panel_config_func: (level, panel_key) => ({
-    depends_on: ['orgTransferPaymentsRegion'],
-    calculate: function(subject, info, options){
-      const { orgTransferPaymentsRegion } = this.tables;
-      return calculate_common(subject, orgTransferPaymentsRegion);
-    },
-    
-    render(render_args){
-      const {
-        calculations,
-        footnotes,
-        sources,
-      } = render_args;
-      const { panel_args } = calculations;
-      const { data } = panel_args;
+class TPMap extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      show_per_capita: false,
+      loading: true,
+      population: {},
+    };
+  }
+  componentDidMount(){
+    loadPopulation()
+      .then( 
+        (transformed) => this.setState({
+          loading: false,
+          population: transformed,
+        })
+      );
+  }
+  render(){
+    const { calculations, footnotes, sources } = this.props;
+    const { show_per_capita, loading, population } = this.state;
+
+    if (loading) {
+      return (
+        <div style = {{position: "relative", height: "80px", marginBottom: "-10px"}}>
+          <SpinnerWrapper/>
```

Comment:
The `SpinnerWrapper` component tags a `config_name` prop. See `src/components/LeafSpinner.js` for the options, but I think this one should be using the "sub_route" config here (or possibly "tabbed_content", that's what the other panels with their own spinners use, but they're technically also all tabbed content).

Reply:
Added the correct sub_route to the args
Done